transformer_engine/pytorch/optimizers/fused_sgd.py

```diff
@@ -74,15 +79,16 @@ class FusedSGD(Optimizer):
 
     def __init__(
         self,
-        params,
-        lr=required,
-        momentum=0,
-        dampening=0,
-        weight_decay=0,
-        nesterov=False,
+        params: Iterable[torch.nn.Parameter | dict],
+        lr: float | Any = required,
+        momentum: float = 0.0,
+        dampening: float = 0.0,
+        weight_decay: float = 0.0,
+        nesterov: bool = False,
+        *,
         wd_after_momentum=False,
```

Comment:
Why not add the type info to those as well?

Reply:
I prioritized options that show up in the PyTorch API: https://pytorch.org/docs/stable/generated/torch.optim.SGD.html#torch.optim.SGD
Otherwise I consider it out of scope for this PR.